### ISSUE TITLE
How to improve performance holistic graph on windows10?

### ISSUE THREAD

**Issue by sewonist** [homuler/MediaPipeUnityPlugin#244]
Hi All,

I've made app what is translator for deaf person with mediapipe. Origin was made with python. And I migration it to Unity for more features. I tested holistic graph with plugin. It works however performance is worst than python. I don't know why delay it. like below image.

![image](https://user-images.githubusercontent.com/352759/129333836-c1a281a0-327b-4d68-ad3c-1d7dcd4204ad.png)

I feel delay time almost over 1sec. So I changed code PushInput slowly. 

```
if (delta > 1f / 20f)
{
  graph.PushInput(nextFrame).AssertOk();  
  graph.RenderOutput(webCamScreenController, nextFrame);
  delta = 0;
}
```

![image](https://user-images.githubusercontent.com/352759/129334200-5aed8e78-d336-42b3-a49e-d7e09ac1816a.png)

It is better than before although after several seconds. It was made delay, too. I don't need high speed. I just want to exactly same output data and video. Anyone has other idea?

Thanks,

**Comment by Joel-Ee** [comment]
How could you compile plugin in windows 10 successfully? We also try as tutorials but failed. Here is our issue: https://github.com/homuler/MediaPipeUnityPlugin/issues/241
maybe we can communicate more about unity performance

**Comment by sewonist** [comment]
@Joel-Ee I just follow tutorial. (https://github.com/homuler/MediaPipeUnityPlugin/#windows) I don't have any problems.

**Comment by sewonist** [comment]
I found one of solutions. I think it's not really good idea but it works for me. I don't know why it was starting slow after lost landmarks. slower and slower. My solution is reset poller every 1sec. I insert below code to OutputStreamPoller.cs

```
    public void Reset()
    {
      UnsafeNativeMethods.mp_OutputStreamPoller__Reset(mpPtr);
    }
```

And I have other question. I really wonder why value of isPoseDetectionPresent is always false? It's normal?

**Comment by jsaon-L** [comment]
I also delay .i use 0.5.3 is ok,but 0.6.2 is faster than 0.5.3

**Comment by sewonist** [comment]
> I also delay .i use 0.5.3 is ok,but 0.6.2 is faster than 0.5.3

What? It happen depend on release? Thanks you for great information. I'm going to test other release.

**Comment by jsaon-L** [comment]
https://github.com/google/mediapipe/issues/2379

**Comment by ROBYER1** [comment]
Thanks for linking my issue with Google here, finding the same latency after a short time running the Pose detection tracking as well as Holistic and not just Holistic. Has anyone found any new fixes or optimizations for this?

**Comment by ROBYER1** [comment]
> I found one of solutions. I think it's not really good idea but it works for me. I don't know why it was starting slow after lost landmarks. slower and slower. My solution is reset poller every 1sec. I insert below code to OutputStreamPoller.cs
> 
> ```
>     public void Reset()
>     {
>       UnsafeNativeMethods.mp_OutputStreamPoller__Reset(mpPtr);
>     }
> ```
> 
> And I have other question. I really wonder why value of isPoseDetectionPresent is always false? Is it normal?

Using this on device causes occasional crashing on IOS/Android :/

**Comment by ROBYER1** [comment]
I just rolled back to Mediapipe 0.8.4  here and rebuilt as it was the most stable. https://github.com/homuler/MediaPipeUnityPlugin/commit/24576ce968c05c748c1ed56f82866c30ea9190d4 

Seems in versions after where they added 3d pose points, a lot of latency was added

**Comment by homuler** [comment]
Fixed by #296
